### ISSUE TITLE
Bug: Movement preview not correctly accounting for different speeds when embarking

### ISSUE THREAD

**Issue by The-Talik** [yairm210/Unciv#12280]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Game Version

4.13.14

### Describe the bug

When calculating a path, the pathfinder assumes the unit's current speed will be used for both embarked and land travel.  This leads to bad/slow routes, as well as incorrect destination turn estimate.

### Steps to Reproduce

Get any upgrade that gives embarked units +x to their embarked speed.
Put unit on land, view path that goes through water.  Land speed is always used.
Put unit in water, view path that goes through land.  Embarked speed is always used.

### Operating System

Android

**Comment by yairm210** [comment]
This is much more complicated than I thought it was going to be.
The problem lies in the fact that embarked units get their extra movement points from *uniques* in techs, which filter on "embarked units". 
The problem is, that unit filters are *text only*, and that - for example - "embarked" checks the current state of the unit.
Changing this to reflect the true "future" values would mean turning the filter into one that can accept how a unit *will* be and not only what it *is*, and that's beyond the scope of what I'm willing to do for this now.

**Comment by The-Talik** [comment]
Yeah, honestly this is a pretty minor issue.  There are better low hanging fruit, like the continue button.
Thanks for looking into it!
